pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -2206,14 +2209,22 @@ func handleOnboardStatusImpl(ctxArg interface{}, key string,
 		return
 	}
 	log.Noticef("Device UUID changed from %s to %s", devUUID, status.DeviceUUID)
+	oldUUID := devUUID
 	devUUID = status.DeviceUUID
 	if zedcloudCtx != nil {
 		zedcloudCtx.DevUUID = devUUID
 	}
 	// Make sure trigger function isn't going to trip on a nil pointer
 	if ctx.getconfigCtx != nil && ctx.getconfigCtx.zedagentCtx != nil &&
 		ctx.getconfigCtx.subAppInstanceStatus != nil {
-
+		if zedcloudCtx != nil && oldUUID != nilUUID {
+			// remove old deferred attest if exists
+			zedcloud.RemoveDeferred(zedcloudCtx, "attest:"+oldUUID.String())
```

Comment:
Wouldn't this be overwritten when we send a new attest request?

Reply:
They have different keys, so, no, they will not override.
Also we do not send them again, we send it only when our device certificates are in place.

Reply:
OK